client/src/app/pages/stock-detail/stock-detail.component.ts

```diff
@@ -8,30 +8,62 @@ import {
 } from '@angular/router';
 import { filter } from 'rxjs/operators';
 
+export interface ChartDataOptions {
+  range: string;
+  interval: string;
+}
+
 @Component({
   selector: 'app-stock-detail',
   templateUrl: './stock-detail.component.html',
   styleUrls: ['./stock-detail.component.scss'],
 })
 export class StockDetailComponent implements OnInit {
+  stockHistoricalData$ = this.storeFacade.stockHistoricalDataLoaded$;
   stockInfo$ = this.storeFacade.currentStockLoaded$;
   marketClock$ = this.storeFacade.currentMarketClock$;
   userInfo$ = this.storeFacade.currentUser$;
+  showStockChart = false;
+  ticker = '';
+  chartRange = '1d';
+  chartInterval = '5m';
   constructor(
     private storeFacade: StoreFacadeService,
     private route: ActivatedRoute,
     private router: Router
-  ) { }
+  ) {}
 
   ngOnInit(): void {
-    let ticker = this.route.snapshot.paramMap.get('ticker');
-    this.storeFacade.loadCurrentStock(ticker);
+    this.stockHistoricalData$.subscribe((hist) => {
+      if (!!hist) {
+        this.showStockChart = true;
+      }
+    });
+    this.ticker = this.route.snapshot.paramMap.get('ticker');
+    this.storeFacade.loadCurrentStock(this.ticker);
+    this.storeFacade.loadCurrentStockHistoricalData(
+      this.ticker,
+      this.chartRange,
+      this.chartInterval
+    );
     this.storeFacade.loadMarketClock();
     this.router.events
       .pipe(filter((event: RouterEvent) => event instanceof NavigationEnd))
       .subscribe(() => {
-        ticker = this.route.snapshot.paramMap.get('ticker');
-        this.storeFacade.loadCurrentStock(ticker);
+        this.ticker = this.route.snapshot.paramMap.get('ticker');
+        this.storeFacade.loadCurrentStock(this.ticker);
+        this.storeFacade.loadCurrentStockHistoricalData(
+          this.ticker,
+          this.chartRange,
+          this.chartInterval
+        );
       });
   }
+  changeChartRangeInterval(chartViewOptions: ChartDataOptions): void {
```

Comment:
Is it possible to pass two strings (range, interval) to this method, instead of having an interface for chartDataOptions?

Reply:
ChartDataOptions is actually from the library so no neeed to change